src/test/util/setup_common.cpp

```diff
@@ -100,9 +104,9 @@ BasicTestingSetup::BasicTestingSetup(const std::string& chainName, const std::ve
     {
         SetupServerArgs(*m_node.args);
         std::string error;
-        const bool success{m_node.args->ParseParameters(arguments.size(), arguments.data(), error)};
-        assert(success);
-        assert(error.empty());
+        if (!m_node.args->ParseParameters(arguments.size(), arguments.data(), error)) {
+            throw std::runtime_error{error};
```

Comment:
For me, `test_bitcoin --run_test=addrman_tests -- -foo` will only report an invalid parameter for the first subtest - subsequent tests of the suite will assert again in `ArgsManager::AddArg`.

I think the reason for this is that if this throws an unhandled exception, the destructor of BasicTestingSetup won't be called and gArgs won't be cleared.

Reply:
Right, I added a call to `ClearArgs()` to undo the `SetupServerArgs()` call.

This code is a bit fishy because it initializes a global variable (`gArgs`) from the constructor of `BasicTestingSetup` and clears it from the destructor...